src/socket.rs

```diff
@@ -206,22 +208,30 @@ impl Socket {
         let _ = inner.buffer_pool.set_active(false);
         gst_debug!(SOCKET_CAT, obj: &inner.element, "Socket shut down");
     }
+
+    pub fn set_do_timestamp(&mut self, timestamp: bool) {
+        let mut inner = self.0.lock().unwrap();
+        inner.do_timestamp = timestamp;
+    }
 }
 
-impl Drop for SocketInner {
+impl<T: SocketRead + 'static> Drop for SocketInner<T> {
     fn drop(&mut self) {
         assert_eq!(self.state, SocketState::Unscheduled);
     }
 }
 
-struct SocketStream(Socket, Option<gst::MappedBuffer<gst::buffer::Writable>>);
+struct SocketStream<T: SocketRead + 'static>(
+    Arc<Mutex<SocketInner<T>>>,
```

Comment:
Why don't we store a `Socket` in here anymore, how does it differ?

Reply:
Socket structure changed Socket<T> structure and the clone() function was returned as a reference type.

Fixed 3deb93a